internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -907,35 +885,38 @@ func (a *Adapter) prepareGroupSnapshot(application *applicationapiv1alpha1.Appli
 	for _, applicationComponent := range *applicationComponents {
 		var isPRMROpened bool
 		applicationComponent := applicationComponent // G601
-		snapshots, err := a.loader.GetMatchingComponentSnapshotsForComponentAndPRGroupHash(a.context, a.client, a.snapshot, applicationComponent.Name, prGroupHash)
-		if err != nil {
-			a.logger.Error(err, "Failed to fetch Snapshots for component", "component.Name", applicationComponent.Name)
-			return nil, nil, err
-		}
 
-		sortedSnapshots := gitops.SortSnapshots(*snapshots)
-		// find the latest component snapshot created for open PR/MR
-		for _, snapshot := range sortedSnapshots {
-			snapshot := snapshot
-			// find the built image for pull/merge request build PLR from the latest opened pull request component snapshot
-			isPRMROpened, err = a.status.IsPRMRInSnapshotOpened(a.context, &snapshot)
+		if slices.Contains(componentsToCheck, applicationComponent.Name) {
```

Comment:
Maybe we could, instead of nesting the logic within this `if` condition, just call `continue` if this condition is not true, e.g:
```
if !slices.Contains(componentsToCheck, applicationComponent.Name) {
    continue
}
snapshots, err := a.loader.GetMatchingComponentSnapshotsForComponentAndPRGroupHash(a.context, a.client, a.snapshot, applicationComponent.Name, prGroupHash)
(...etc...)
```

Reply:
If not containing that component, we still need to get snapshotComponent data from GCL for this component.

Reply:
Ah, ok that's a good point.

Reply:
Maybe in that case everything that's contained within this `if ` condition could be extracted to a separate function since its logic is contained to finding out if a PR/MR is opened? 
Something like `findSnapshotWithOpenedPR()` or similar?

Reply:
Yes, I tried it yesterday like [this](https://github.com/konflux-ci/integration-service/commit/39febc38d38f3322267ed70af08c5205b74b7601#diff-cbf7e6d58cc12a67b910b5b9108ba68d206e75cc8560ca0ab2b19e2d34415702R1045-R1068), then I checked the complexity but I found the complexity doesn't change because I checked EnsureGroupSnapshotExist() wrongly not this prepareGroupSnapshot.

I am also thinking to use pointer for `groupSnapshot, componentSnapshotInfos` to have code like ` if ok {func_a} else {func_b}` without returning groupSnapshot, componentSnapshotInfos

Reply:
`findSnapshotWithOpenedPR()` is a good name, let me update. Thank you.

Reply:
That looks great, thanks!